File: tests/statistical_tests/spot_checks/proportion-two-sample.test.ts
import { stats } from "../../../src/dataframe/mod.ts";

Deno.test("Two-Sample Proportion Test Spot Check", () => {
  console.log("\n📊 Two-Sample Proportion Test");
  console.log("-".repeat(80));

  const group1 = [true, true, false, true, true, true, false, true];
  const group2 = [false, true, false, false, true, false, false, true];

  const result = stats.test.proportion.twoSample({
    data1: group1,
    data2: group2,
    alternative: "two-sided",
  });

  const successes1 = group1.filter((x) => x).length;
  const successes2 = group2.filter((x) => x).length;

  console.log("Group 1:", group1);
  console.log("Group 1 successes:", successes1, "/", group1.length);
  console.log("Group 2:", group2);
  console.log("Group 2 successes:", successes2, "/", group2.length);
  console.log("Test statistic:", result.test_statistic.value);
  console.log("p-value:", result.p_value);
  console.log("CI:", [
    result.confidence_interval.lower,
    result.confidence_interval.upper,
  ]);
});

console.log("\n" + "=".repeat(80));
console.log("TWO-SAMPLE PROPORTION TEST SPOT CHECK");
console.log("=".repeat(80));

File: packages/testing/statistical_tests/spot_checks/proportion-two-sample.test.R
#!/usr/bin/env Rscript

cat("\n📊 Two-Sample Proportion Test\n")
cat(paste(rep("-", 80), collapse = ""), "\n")

successes <- c(6, 3)
n <- c(8, 8)

# Two-sample proportion test
result <- prop.test(successes, n, alternative = "two.sided", conf.level = 0.95, correct = TRUE)

cat("Group 1 successes:", successes[1], "/", n[1], "\n")
cat("Group 2 successes:", successes[2], "/", n[2], "\n")
cat("Chi-squared statistic:", result$statistic, "\n")
cat("p-value:", result$p.value, "\n")
cat("CI lower:", result$conf.int[1], "\n")
cat("CI upper:", result$conf.int[2], "\n")
cat("Sample estimates:", result$estimate, "\n")

cat("\n", paste(rep("=", 80), collapse = ""), "\n")
cat("TWO-SAMPLE PROPORTION TEST SPOT CHECK\n")
cat(paste(rep("=", 80), collapse = ""), "\n")
